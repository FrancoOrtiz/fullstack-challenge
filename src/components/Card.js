import React from 'react'

export const Card = () => {
  return (
    <div className="flex w-80 h-40 justify-between my-10">
      <div className="h-full w-40 bg-gray-500">
        <img src=""></img>
      </div>
      <span className="mt-2 text-xl text-gray-500">Pikachu</span>
    </div>
  )
}
