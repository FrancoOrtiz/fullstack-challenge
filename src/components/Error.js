import React from "react";

export const Error = ({ message }) => {
  return (
    <div className="bg-red-200 border-2 border-red-500 rounded-lg flex items-center justify-center text-red-600 m-auto mb-4 w-full h-14">
      {message}
    </div>
  );
};
