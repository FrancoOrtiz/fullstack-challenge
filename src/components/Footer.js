import React from "react";

export const Footer = () => {
  return (
    <footer className="w-4/5 m-auto flex justify-between mt-10 border-t-2 border-gray-400 items-center h-20 pl-10 pr-20">
      <p>Hecho por Franco</p>
      <a
        href="https://github.com/FrancoOrtiz/"
        target="blank"
        className="w-1/4 py-2 shadow-button text-center text-xl rounded-xl bg-gray-300 font-bold border-black border-2"
      >
        Link a mi repo
      </a>
    </footer>
  );
};
