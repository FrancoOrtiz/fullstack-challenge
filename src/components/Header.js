import React from "react";

export const Header = () => {
  return (
    <header className="flex flex-col gap-3 w-4/5 m-auto">
      <h1 className="text-black text-6xl font-bold">Pokemon finder</h1>
      <p className="text-gray-500 text-xl">
        El que quiere Pokemons, que los busque.
      </p>
    </header>
  );
};
