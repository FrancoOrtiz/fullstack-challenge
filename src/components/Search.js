import React, {useContext} from 'react'
import { DataContext } from '../context/DataContext'
import { SearchBar } from './SearchBar'
import { SearchResult } from './SearchResult'

export const Search = () => {
  
  const {pokemons} = useContext(DataContext);

  return (
    <div className="w-4/5 m-auto">
        <SearchBar></SearchBar>
        <SearchResult pokemons={pokemons}></SearchResult>
    </div>
  )
}
