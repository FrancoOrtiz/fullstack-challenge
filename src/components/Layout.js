import React, {Fragment} from 'react'
import { Footer } from './Footer'
import { Header } from './Header'

export const Layout = ({children}) => {
  return (
    <Fragment>
        <Header/>
            {children}
        <Footer/>
    </Fragment>
  )
}
