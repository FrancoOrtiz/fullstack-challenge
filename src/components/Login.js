import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { useAuth } from "../context/authContext";
import { Header } from "./Header";
import { Error } from "./Error";

export const Login = () => {
  const [user, setUser] = useState({
    email: "",
    password: "",
  });
  const [error, setError] = useState();
  const { login, loginWithGoogle, resetPassword } = useAuth();
  const navigate = useNavigate();

  const handleChange = ({ target: { name, value } }) => {
    setUser({ ...user, [name]: value });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await login(user.email, user.password);
      navigate("/");
    } catch (error) {
      setError(error.message);
    }
  };

  const handleGoogleLogin = async () => {
    try {
      await loginWithGoogle();
      navigate("/");
    } catch (error) {
      setError(error.code);
    }
  };

  const handleForgot = async () => {
    if (!user.email) {
      setError("Ingrese su correo");
    } else {
      try {
        await resetPassword(user.email);
        setError("Te hemos enviado un correo para que recuperes tu contraseña");
      } catch (error) {
        setError(error.message);
      }
    }
  };

  const navRegister = () => {
    navigate("/register");
  };

  return (
    <div className="mt-32 w-4/5 lg:w-2/4 m-auto">
      {error && <Error message={error} />}

      <form
        onSubmit={handleSubmit}
        className=" pt-10 pb-6 flex flex-col text-center gap-6 bg-white rounded-xl shadow-lg shadow-slate-400"
      >
        <Header></Header>
        <div className="w-4/5 m-auto flex flex-col gap-1 my-0">
          <label htmlFor="email">Email</label>
          <input
            type="email"
            name="email"
            placeholder="youremail@company.ltd"
            className="border-solid border-2 rounded-md border-gray-500 h-10 pl-2 w-2/3 lg:w-1/2 m-auto"
            onChange={handleChange}
          />
        </div>

        <div className="w-4/5 m-auto flex flex-col gap-1 my-0">
          <label htmlFor="password">Password</label>
          <input
            type="password"
            name="password"
            id="password"
            placeholder="******"
            className="border-solid border-2 rounded-md border-gray-500 h-10 pl-2 w-2/3 lg:w-1/2 m-auto"
            onChange={handleChange}
          />
        </div>

        <div className="w-full flex-col items-center flex pt-6 mt-6 border-solid border-t-2 border-gray-200">
          <button className="w-44 h-14 bg-blue-500 font-bold text-white rounded-lg text-lg hover:bg-transparent border-2 border-blue-500 hover:text-blue-500 transition-all">
            Login
          </button>
          <button
            type="button"
            onClick={handleGoogleLogin}
            className="w-52 h-14 border-2 font-bold border-black rounded-lg text-lg mt-4 "
          >
            Login with Google
          </button>
        </div>
      </form>
      <div className="mt-4 flex w-full justify-between">
        <button
          type="button"
          className="text-blue-400 mt-4 hover:text-blue-600"
          onClick={navRegister}
        >
          No tienes un usuario? Clickea aqui
        </button>
        <button
          type="button"
          className="text-blue-400 mt-4 hover:text-blue-600"
          onClick={handleForgot}
        >
          Olvidaste tu contraseña?
        </button>
      </div>
    </div>
  );
};
