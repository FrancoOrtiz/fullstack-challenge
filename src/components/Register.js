import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { useAuth } from "../context/authContext";
import { Error } from "./Error";
import { Header } from "./Header";

export const Register = () => {
  const [user, setUser] = useState({
    email: "",
    password: "",
  });
  const [error, setError] = useState();
  const { signup } = useAuth();
  const navigate = useNavigate();

  const handleChange = ({ target: { name, value } }) => {
    setUser({ ...user, [name]: value });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await signup(user.email, user.password);
      navigate("/");
    } catch (error) {
      setError(error.message);
    }
  };
  const navLogin = () => {
    navigate("/login");
  };

  return (
    <div className="mt-32 w-4/5 lg:w-2/4 m-auto">
      {error && <Error message={error} />}

      <form
        onSubmit={handleSubmit}
        className=" pt-10 pb-6 flex flex-col text-center gap-6 bg-white rounded-xl shadow-lg shadow-slate-400"
      >
        <Header></Header>
        <div className="w-4/5 m-auto flex flex-col gap-1 my-0">
          <label htmlFor="email">Email</label>
          <input
            type="email"
            name="email"
            placeholder="youremail@company.ltd"
            className="border-solid border-2 rounded-md border-gray-500 h-10 pl-2 w-2/3 lg:w-1/2 m-auto"
            onChange={handleChange}
          />
        </div>

        <div className="w-4/5 m-auto flex flex-col gap-1 my-0">
          <label htmlFor="password">Password</label>
          <input
            type="password"
            name="password"
            id="password"
            placeholder="******"
            className="border-solid border-2 rounded-md border-gray-500 h-10 pl-2 w-2/3 lg:w-1/2 m-auto"
            onChange={handleChange}
          />
        </div>

        <div className="w-full flex-col items-center flex pt-6 mt-6 border-solid border-t-2 border-gray-200">
          <button className="w-44 h-14 font-bold bg-blue-500 text-white rounded-lg text-lg hover:bg-transparent border-2 border-blue-500 hover:text-blue-500 transition-all">
            Register
          </button>
        </div>
      </form>
      <button
        type="button"
        className="text-blue-400 mt-6 hover:text-blue-600"
        onClick={navLogin}
      >
        Ya tienes un usuario? Clickea aqui
      </button>
    </div>
  );
};
