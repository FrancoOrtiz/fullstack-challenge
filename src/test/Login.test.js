const { Login } = require("components/Login");
const { authContext } = require("context/authContext");
const { BrowserRouter } = require("react-router-dom");
import { fireEvent, render, waitFor } from "@testing-library/react";
import { Error } from "components/Error";
import renderer from "react-test-renderer";

describe("Tests on Login.js", () => {
  let wrapper;
  const mockLogin = jest.fn();
  const mockLoginGoogle = jest.fn();
  const mockReset = jest.fn();

  test("should be render", () => {
    wrapper = renderer
      .create(
        <authContext.Provider value={{}}>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </authContext.Provider>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  test("should be call login function", () => {
    wrapper = render(
      <authContext.Provider value={{ login: mockLogin }}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </authContext.Provider>
    );
    const btn = wrapper.getByText("Login");
    expect(btn).toBeDefined();

    waitFor(() => fireEvent.click(btn));
    expect(mockLogin).toBeCalledTimes(1);
  });

  test("should be call login with the arguments email and password", () => {
    wrapper = render(
      <authContext.Provider value={{ login: mockLogin }}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </authContext.Provider>
    );

    const email = wrapper.getByPlaceholderText("youremail@company.ltd");
    fireEvent.change(email, { target: { value: "tests@email.com" } });
    const password = wrapper.getByPlaceholderText("******");
    fireEvent.change(password, { target: { value: "123456" } });

    const btn = wrapper.getByText("Login");
    expect(btn).toBeDefined();

    waitFor(() => fireEvent.click(btn));
    expect(mockLogin).toHaveBeenCalledWith("tests@email.com", "123456");
  });

  test("should show Error Component if password is wrong", () => {
    wrapper = render(
      <authContext.Provider value={{ login: mockLogin }}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </authContext.Provider>
    );

    const email = wrapper.getByPlaceholderText("youremail@company.ltd");
    fireEvent.change(email, { target: { value: "tests@email.com" } });
    const password = wrapper.getByPlaceholderText("******");
    fireEvent.change(password, { target: { value: "1234567" } });

    const btn = wrapper.getByText("Login");
    expect(btn).toBeDefined();

    waitFor(() => fireEvent.click(btn));
    expect(<Error />).toBeDefined();
  });

  test("should show Error Component if the user is not found", () => {
    wrapper = render(
      <authContext.Provider value={{ login: mockLogin }}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </authContext.Provider>
    );

    const email = wrapper.getByPlaceholderText("youremail@company.ltd");
    fireEvent.change(email, { target: { value: "imNotRegister@email.com" } });
    const password = wrapper.getByPlaceholderText("******");
    fireEvent.change(password, { target: { value: "1234567" } });

    const btn = wrapper.getByText("Login");
    expect(btn).toBeDefined();

    waitFor(() => fireEvent.click(btn));
    expect(<Error />).toBeDefined();
  });

  test("should be call loginWithGoogle function", () => {
    wrapper = render(
      <authContext.Provider value={{ loginWithGoogle: mockLoginGoogle }}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </authContext.Provider>
    );

    const btn = wrapper.getByText("Login with Google");
    expect(btn).toBeDefined();

    waitFor(() => fireEvent.click(btn));
    expect(mockLoginGoogle).toBeCalledTimes(1);
  });

  test("should be call resetPassword function", () => {
    wrapper = render(
      <authContext.Provider value={{ resetPassword: mockReset }}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </authContext.Provider>
    );

    const btn = wrapper.getByText("Olvidaste tu contraseña?");
    expect(btn).toBeDefined();

    const email = wrapper.getByPlaceholderText("youremail@company.ltd");
    fireEvent.change(email, { target: { value: "tests@email.com" } });

    waitFor(() => fireEvent.click(btn));
    expect(mockReset).toBeCalledTimes(1);
    expect(
      "Te hemos enviado un correo para que recuperes tu contraseña"
    ).toBeDefined();
  });

  test("should ask the email", () => {
    wrapper = render(
      <authContext.Provider value={{ resetPassword: mockReset }}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </authContext.Provider>
    );

    const btn = wrapper.getByText("Olvidaste tu contraseña?");
    expect(btn).toBeDefined();

    const email = wrapper.getByPlaceholderText("youremail@company.ltd");
    fireEvent.change(email, { target: { value: "" } });

    waitFor(() => fireEvent.click(btn));
    expect("Ingrese su correo").toBeDefined();
  });
});
