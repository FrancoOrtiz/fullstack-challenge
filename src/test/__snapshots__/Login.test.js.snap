// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on Login.js should be render 1`] = `
<div
  className="mt-32 w-4/5 lg:w-2/4 m-auto"
>
  <form
    className=" pt-10 pb-6 flex flex-col text-center gap-6 bg-white rounded-xl shadow-lg shadow-slate-400"
    onSubmit={[Function]}
  >
    <header
      className="flex flex-col gap-3 w-4/5 m-auto"
    >
      <h1
        className="text-black text-6xl font-bold"
      >
        Pokemon finder
      </h1>
      <p
        className="text-gray-500 text-xl"
      >
        El que quiere Pokemons, que los busque.
      </p>
    </header>
    <div
      className="w-4/5 m-auto flex flex-col gap-1 my-0"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="border-solid border-2 rounded-md border-gray-500 h-10 pl-2 w-2/3 lg:w-1/2 m-auto"
        name="email"
        onChange={[Function]}
        placeholder="youremail@company.ltd"
        type="email"
      />
    </div>
    <div
      className="w-4/5 m-auto flex flex-col gap-1 my-0"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="border-solid border-2 rounded-md border-gray-500 h-10 pl-2 w-2/3 lg:w-1/2 m-auto"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="******"
        type="password"
      />
    </div>
    <div
      className="w-full flex-col items-center flex pt-6 mt-6 border-solid border-t-2 border-gray-200"
    >
      <button
        className="w-44 h-14 bg-blue-500 font-bold text-white rounded-lg text-lg hover:bg-transparent border-2 border-blue-500 hover:text-blue-500 transition-all"
      >
        Login
      </button>
      <button
        className="w-52 h-14 border-2 font-bold border-black rounded-lg text-lg mt-4 "
        onClick={[Function]}
        type="button"
      >
        Login with Google
      </button>
    </div>
  </form>
  <div
    className="mt-4 flex w-full justify-between"
  >
    <button
      className="text-blue-400 mt-4 hover:text-blue-600"
      onClick={[Function]}
      type="button"
    >
      No tienes un usuario? Clickea aqui
    </button>
    <button
      className="text-blue-400 mt-4 hover:text-blue-600"
      onClick={[Function]}
      type="button"
    >
      Olvidaste tu contrase√±a?
    </button>
  </div>
</div>
`;
